features/support/steps.js

```diff
@@ -91,10 +91,28 @@ When('the user clicks the save button', function () {
   });
 });
 
-Then(/the report (?:will be|is) displayed/, { timeout: 60 * 1000 }, function () {
-  // Wait for a second for the loading ind to disappear
-  return this.app.client.waitUntilWindowLoaded().getText('#output')
-    .should.eventually.not.equal('');
+When('the user clicks the settings button', function () {
+  return this.app.client.$('#appSettingsButton').then((element) => element.click());
+});
+
+When('the user selects {string} from the Default Gherkin Dialect drop-down menu', function (dialect) {
+  return this.app.client.$('#dialectSelection').then((element) => element.selectByVisibleText(dialect));
+});
+
+Then(/the report (?:will be|is) (displayed|hidden)/, { timeout: 60 * 1000 }, async function (reportStatus) {
+  const displayed = reportStatus === 'displayed';
+  const outputElem = await this.app.client.$('#output');
+  if (displayed) {
+    const outputText = await outputElem.getText();
+    expect(outputText).to.not.equal('');
+  }
+  const isDisplayed = await outputElem.isDisplayed();
+  expect(isDisplayed).to.equal(displayed);
+});
+
+Then(/the settings view will be (displayed|hidden)/, function (settingsStatus) {
+  const displayed = settingsStatus === 'displayed';
+  return this.app.client.$('#appSettings').then((settingsElem) => settingsElem.isDisplayed().should.eventually.equal(displayed));
```

Comment:
```suggestion
  return this.app.client.$('#appSettings')
    .then((settingsElem) => settingsElem.isDisplayed().should.eventually.equal(displayed));
```

Reply:
Is there anything preventing us from doing this one and those further down that use promise chains as async/await functions instead of promise chains?

Reply:
Short answer is no, I don't think so.  Way back when we started I think I was having trouble getting the async/await to play nicely with the Webdriverio apis, but now that those are upgraded (and I have half of a clue about async/await) we could do that everywhere I think.  Big difference would be that we would need to have a lot more lines/variables since we could not chain things....